independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java

```diff
@@ -421,7 +421,8 @@ private void processDecorator(DecoratorInfo decorator, BeanInfo bean, Type provi
         ClassCreator delegateSubclass = delegateSubclassBuilder.build();
 
         Map<MethodDescriptor, DecoratorInfo> nextDecorators = bean.getNextDecorators(decorator);
-        List<DecoratorInfo> decoratorParameters = new ArrayList<>(nextDecorators.values());
+        Collection<DecoratorInfo> nextDecoratorsValues = nextDecorators.values();
+        List<DecoratorInfo> decoratorParameters = new ArrayList<>(new HashSet<>(nextDecoratorsValues));
```

Comment:
This is actually a different fix, or a small optimization - we don't need to create a "next decorator" field for each decorated method if the same decorator is used. The same reference would be set anyway.